app/gui/language/span-tree/src/generate.rs

```diff
@@ -818,11 +817,53 @@ fn generate_trailing_expected_arguments(
     })
 }
 
+/// A single child node produced out of the lambda argument and the `->` token.
+#[derive(Debug)]
+struct FoldedLambdaArguments {
+    /// Both the lambda argument and the `->` token, as a single [`node::Kind::Token`] node.
+    child:          node::Child,
+    /// The number of tree nodes that were folded into the `child`.
+    nodes_replaced: usize,
+}
 
-
-// =========================
-// === SpanTree for Tree ===
-// =========================
+/// Fold the lambda arguments into a single [`node::Kind::Token`] node.
+/// It is needed to ignore lambda arguments as connection targets, but still generate a valid
+/// SpanTree from the lambda body.
+fn fold_lambda_arguments(
+    tree: &ast::Tree<Ast>,
+    context: &impl Context,
+) -> Result<FoldedLambdaArguments, failure::Error> {
+    let is_arrow = |span_info| matches!(span_info, SpanSeed::Token(ast::SpanSeedToken { token }) if token == "->");
+    let arrow_index = tree.span_info.iter().cloned().position(is_arrow).unwrap_or(0);
+    let bytes_till_body = tree
+        .span_info
+        .iter()
+        .take(arrow_index + 1)
+        .map(|raw_span_info| match raw_span_info {
+            SpanSeed::Space(ast::SpanSeedSpace { space }) => Ok(*space),
+            SpanSeed::Token(ast::SpanSeedToken { token }) => Ok(token.len()),
+            SpanSeed::Child(ast::SpanSeedChild { node }) => {
+                let kind = node::Kind::argument().with_removable(false);
+                let node = node.generate_node(kind, context);
+                node.map(|node| node.size.as_usize())
+            }
```

Comment:
Do I understand correctly that we generate node only to discard it and read its size?

Are there no better options for measuring new AST node size? Like `repr_len` or `char_count` in the old AST? I believe generating span tree does much more things.

Reply:
Yep, `repr_len` is better here, thanks for noting. I changed the code.